tests/chains/Bitcoin/TWBitcoinSigningTests.cpp

```diff
@@ -175,19 +180,25 @@ TEST(BitcoinSigning, SignBRC20TransferCommit) {
     *utxo.mutable_out_point() = out;
 
     Proto::TransactionPlan plan;
+
     auto& utxo1 = *plan.add_utxos();
     utxo1.set_amount(brcInscribeAmount);
     utxo1.set_script(outputInscribe.script());
     utxo1.set_variant(Proto::TransactionVariant::BRC20TRANSFER);
+    *utxo1.mutable_out_point() = emptyTxOutPoint.proto();
 
     auto& utxo2 = *plan.add_utxos();
     utxo2.set_amount(forFeeAmount);
     utxo2.set_script(inputP2wpkh.bytes.data(), inputP2wpkh.bytes.size());
     utxo2.set_variant(Proto::TransactionVariant::P2WPKH);
+    *utxo2.mutable_out_point() = emptyTxOutPoint.proto();
 
     *input.mutable_plan() = plan;
-    Proto::SigningOutput output;
 
+    ANY_PLAN(input, plan, TWCoinTypeBitcoin);
```

Comment:
Could you please consider these changes?
```c++
TEST(BitcoinSigning, SignBRC20TransferCommit) {
    auto privateKey = parse_hex("e253373989199da27c48680e3a3fc0f648d50f9a727ef17a7fe6a4dc3b159129");
    auto fullAmount = 26400;
    auto minerFee = 3000;
    auto brcInscribeAmount = 7000;
    auto forFeeAmount = fullAmount - brcInscribeAmount - minerFee;
    auto txId = parse_hex("089098890d2653567b9e8df2d1fbe5c3c8bf1910ca7184e301db0ad3b495c88e");

    PrivateKey key(privateKey);
    auto pubKey = key.getPublicKey(TWPublicKeyTypeSECP256k1);
    auto utxoPubKeyHash = Hash::ripemd(Hash::sha256(pubKey.bytes));
    auto inputP2wpkh = TW::Bitcoin::Script::buildPayToWitnessPublicKeyHash(utxoPubKeyHash);
    auto outputInscribe = TW::Bitcoin::Script::buildBRC20InscribeTransfer("oadf", 20, pubKey.bytes);

    Proto::SigningInput input;
    input.set_is_it_brc_operation(true);
    input.add_private_key(key.bytes.data(), key.bytes.size());
    input.set_coin_type(TWCoinTypeBitcoin);
    input.set_amount(fullAmount);
    input.set_byte_fee(19);
    input.set_to_address("1Bp9U1ogV3A14FMvKbRJms7ctyso4Z4Tcx");

    auto& utxo1 = *input.add_utxo();
    utxo1.set_amount(brcInscribeAmount);
    utxo1.set_script(outputInscribe.script());
    utxo1.set_variant(Proto::TransactionVariant::BRC20TRANSFER);

    Proto::OutPoint out1;
    out1.set_index(1);
    out1.set_hash(txId.data(), txId.size());
    *utxo1.mutable_out_point() = out1;

    auto& utxo2 = *input.add_utxo();
    utxo2.set_amount(forFeeAmount);
    utxo2.set_script(inputP2wpkh.bytes.data(), inputP2wpkh.bytes.size());
    utxo2.set_variant(Proto::TransactionVariant::P2WPKH);

    // TODO fill the `out2` with the correct values
    Proto::OutPoint out2;
    out2.set_index(1);
    out2.set_hash(txId.data(), txId.size());
    *utxo2.mutable_out_point() = out2;
    
    Proto::TransactionPlan plan;
    ANY_PLAN(input, plan, TWCoinTypeBitcoin);
    ASSERT_EQ(plan.fee(), 2907);
    // TODO check all fields of the plan.
    
    // Set the plan to the signing input after `ANY_PLAN`.
    *input.mutable_plan() = plan;

    Proto::SigningOutput output;
    ANY_SIGN(input, TWCoinTypeBitcoin);
    ASSERT_EQ(hex(output.encoded()), "02000000000101089098890d2653567b9e8df2d1fbe5c3c8bf1910ca7184e301db0ad3b495c88e0100000000ffffffff02581b000000000000225120e8b706a97732e705e22ae7710703e7f589ed13c636324461afa443016134cc051040000000000000160014e311b8d6ddff856ce8e9a4e03bc6d4fe5050a83d02483045022100a44aa28446a9a886b378a4a65e32ad9a3108870bd725dc6105160bed4f317097022069e9de36422e4ce2e42b39884aa5f626f8f94194d1013007d5a1ea9220a06dce0121030f209b6ada5edb42c77fd2bc64ad650ae38314c8f451f3e36d80bc8e26f132cb00000000");
    ASSERT_EQ(output.transaction_id(), "797d17d47ae66e598341f9dfdea020b04d4017dcf9cc33f0e51f7a6082171fb1");
    ASSERT_EQ(output.error(), Common::Proto::OK);
    // Successfully broadcasted: https://www.blockchain.com/explorer/transactions/btc/797d17d47ae66e598341f9dfdea020b04d4017dcf9cc33f0e51f7a6082171fb1
}
```

I believe `SigningInput::plan` should be set after `ANY_PLAN` only

Reply:
Can you explain what's happening here? 🤔 There should be exactly one P2WPKH input, and two outputs, specifically a BRC20TRANSFER and a P2WPKH (change) output. The resulting plan it constructs is kind of odd, it uses the same inputs as outputs?

```json
{
    "version": 2,
    "locktime": 0,
    "ins": [
        {
            "n": 1,
            "script": {
                "asm": "",
                "hex": ""
            },
            "sequence": 4294967295,
            "txid": "8ec895b4d30adb01e38471ca1019bfc8c3e5fbd1f28d9e7b5653260d89989008",
            "witness": [
                "01b0a0df92f661bc86b436e273c89644ca64f186547a87d45ba6c222f854d21571a16352a6851f775f4e3628a2c563f170ebbcc10198e79b7a4dc8dbe1cf5fc0",
                "",
                "c10f209b6ada5edb42c77fd2bc64ad650ae38314c8f451f3e36d80bc8e26f132cb"
            ]
        },
        {
            "n": 1,
            "script": {
                "asm": "",
                "hex": ""
            },
            "sequence": 4294967295,
            "txid": "8ec895b4d30adb01e38471ca1019bfc8c3e5fbd1f28d9e7b5653260d89989008",
            "witness": [
                "3045022100d34e69314342edd6786bc578265d6f4761aa0a3b2b881f816ba2e3a7742c0220022030e47ecf7e0e7ee8f0620036aaccf4bf780104762cc76336556a1e92e85ed4b601",
                "030f209b6ada5edb42c77fd2bc64ad650ae38314c8f451f3e36d80bc8e26f132cb"
            ]
        }
    ],
    "outs": [
        {
            "n": 0,
            "script": {
                "addresses": [],
                "asm": "OP_1 e8b706a97732e705e22ae7710703e7f589ed13c636324461afa443016134cc05",
                "hex": "5120e8b706a97732e705e22ae7710703e7f589ed13c636324461afa443016134cc05"
            },
            "value": 7000
        },
        {
            "n": 1,
            "script": {
                "addresses": [
                    "bc1quvgm34kal7zke68f5nsrh3k5leg9p2pa2nlgsp"
                ],
                "asm": "OP_0 e311b8d6ddff856ce8e9a4e03bc6d4fe5050a83d",
                "hex": "0014e311b8d6ddff856ce8e9a4e03bc6d4fe5050a83d"
            },
            "value": 16400
        }
    ],
    "hash": "94ccca06f46a1586c1440b10a7e85effff31dfc98c37e4b008091ea006b8d6fb",
    "txid": "94ccca06f46a1586c1440b10a7e85effff31dfc98c37e4b008091ea006b8d6fb"
}
```